docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -20,196 +45,193 @@ The scenarios of defining placement rules in SQL include:
 
 ```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=voter REPLICAS=2,
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=voter REPLICAS=2;
+CREATE TABLE t1 (
+	id INT NOT NULL PRIMARY KEY,
+	b VARCHAR(100)
+) PLACEMENT POLICY=`companystandardpolicy`;
 ```
 
-The same role `voter` is defined in 2 different rules, each of which adds 2 replicas. So it is equivalent to:
+Adding `PLACEMENT POLICY` to an existing table:
 
 ```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="{+zone=sh:2,+zone=bj:2}" ROLE=voter REPLICAS=4;
+CREATE TABLE t1 (
+	id INT NOT NULL PRIMARY KEY,
+	b VARCHAR(100)
+);
+ALTER TABLE t1 PLACEMENT POLICY=`companystandardpolicy`;
 ```
 
-Note that as there may already exist 3 replicas by default, so it will be 7 replicas after executing this statement. So `ADD PLACEMENT POLICY` can be taken as a shortcut for adding replicas to a defined role. In the example above, it can be replaced by `ALTER PLACEMENT POLICY`.
+Expected Behaviors:
+- `CREATE` or `ALTER` and specifying a `PLACEMENT POLICY` that does not exist results in an error: placement policy 'x' is not defined (TBD: see "unresolved questions")
+- Placement policies are globally unique names. Thus, a policy named `companyplacementpolicy` can apply to the db `test` as well as `userdb`. The namespace does not overlap with other DB objects.
+- Placement Policy names are case insensitive, and follow the same rules as tables/other identifiers for length (64 chars) and special characters.
+- The full placement policy can be seen with `SHOW CREATE PLACEMENT POLICY x`. This is useful for shorthand usage by DBAs, and consistent with other database objects.
+- It is possible to update the definition of a placement policy with `ALTER PLACEMENT POLICY x CONSTRAINTS ..;` This is modeled on the statement `ALTER VIEW` (where the view needs to be redefined)
+- The statement `DROP PLACEMENT POLICY` should return an error if the policy is in use by any objects (TBD: see "unresolved questions")
 
-More details of `CONSTRAINTS` option is described in the "Constraints Configuration" section.
+#### Metadata commands
 
-`ADD PLACEMENT POLICY` is implemented by adding one or more placement rules in PD. The statement must wait until the PD returns a message. It can be cancelled by executing `ADMIN CANCEL DDL JOBS` statement.
+Besides `SHOW CREATE PLACEMENT POLICY x`, it should be possible to summarize all placement for a database system. This is beneficial for compliance scenarios.
 
-### Alter placement rules
+#### information_schema.placement_rules
 
-Altering current placement rules can be done by one or more `ALTER PLACEMENT POLICY` clauses:
+A new system table `information_schema.placement_rules` is added to view all placement rules. The table contains such columns:
 
-```sql
-ALTER TABLE table_name
-	ALTER PLACEMENT POLICY CONSTRAINTS=constraints ROLE=role REPLICAS=replicas,
-	...
-```
+* rule_id
+* target ID
+* target name
+* constraints
+* role
+* replicas
```

Comment:
If we make constrants a list of json objects, we can remove role and replicas here.

Reply:
note list of json object is also a json object, and TiDB supports JSON (though not GAed yet, but we can always eat out own dog food)

Reply:
This has been improved in the newer syntax. PTAL again, thx!